src/main/java/CalcServlet.java

```diff
@@ -0,0 +1,125 @@
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.*;
+import java.util.regex.Pattern;
+
+@WebServlet(name = "CalcServlet", urlPatterns = "/calc")
+public class CalcServlet extends HttpServlet {
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+    }
+
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        PrintWriter out = response.getWriter();
+        Map<String, String[]> vars = new HashMap<>(request.getParameterMap());
+        String equation = replaceVars(vars);
+        out.println(evaluate(equation));
+    }
+
+    private String replaceVars(Map<String, String[]> vars) {
+        String equation = vars.remove("equation")[0];
+        for (Map.Entry<String, String[]> param : vars.entrySet()) {
+            String value;
+            if(Character.isAlphabetic(param.getValue()[0].charAt(0))) { // if variable refers to another variable
+                value = vars.get(param.getValue()[0])[0]; // get value of referral variable
+            } else {
+                value = param.getValue()[0];
+            }
+            equation = equation.replace(param.getKey(), value);
+        }
+        System.out.println("[+] Replaced vars with their values.");
+        return equation;
+    }
+
+    private int evaluate(String equation) {
+        List<String> rpn = getPostfixOf(equation); //infix form to postfix form for calc algorithm
+        return getValueOf(rpn);
+    }
+
+    private int getValueOf(List<String> postfix) {
+        Stack<Integer> digs = new Stack<>();
+        Pattern isNum = Pattern.compile("\\d+");
+        for (String token :
+                postfix) {
+            if (isNum.matcher(token).matches()) {
+                digs.push(Integer.parseInt(token));
+            } else {
+                int ans = doOperation(digs.pop(), digs.pop(), token.charAt(0));
+                digs.push(ans);
+            }
+        }
+        System.out.println("[+] Calculated an expression.");
+        return digs.pop();
+    }
+
+    private int doOperation(int rightValue, int leftValue, char operation) {
+        switch (operation) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Switch statements should have a default label](https://app.codacy.com/app/thejerome/IFMO_Servlets_20190915/pullRequest?prid=4635185)